### PR TITLE
Allow local network access from other devices when in development

### LINKED ISSUE #820: Allow access from hosts other than `localhost` during development
## Problem

To be able to access a locally running nuxt server from another device on the same network, you need to either tunnel the request through something like `caddy` or `ngrok` or update the `nuxt.config.js` `server.host` property to `0.0.0.0`. We should just set the the `host` property to `0.0.0.0` automatically in dev. It doesn't hurt the existing workflow and only makes it more flexible for testing across multiple devices.

## Alternatives

Recommend an official method of tunneling requests locally (like `caddy`) and add a `package.json` script for initiating the tunneling.

The idea is to just make this process easier and not require temporary code changes.

## Implementation

- [x] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Fixes

Fixes #820 by @sarayourfriend

## Description

Nuxt documentation about this change here: https://nuxtjs.org/docs/features/configuration#edit-host-and-port

## Testing Instructions

Checkout the branch and run `pnpm dev` then visit the locally running site from another device like a mobile phone or another computer. It should work now. Also visit it at `localhost:8443` like we used to. That should also work.

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [ ] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by zackkrida** [comment]
It's odd that they show the strategy they _don't_ recommend first, but from the docs:

> Although you can modify this in the nuxt.config.js file it is not advised to as it might cause you issues when hosting your site. It is much better to modify the host and port direct in the dev command.

They recommend adding `--hostname '0'` to the `dev` npm script instead.

**Comment by sarayourfriend** [comment]
Oops, I should have kept reading 🙂 I've changed it to use the `hostname` flag. Thanks for finding that Zack!

File: nuxt.config.js [hunk 1]
```diff
@@ -111,7 +111,9 @@ export default {
   },
   srcDir: 'src/',
   modern: 'client',
-  server: { port: process.env.PORT || 8443 },
+  server: {
+    port: process.env.PORT || 8443,
+  },
```
Comment (sarayourfriend) [inline nuxt.config.js:116]:
```suggestion
  server: { port: process.env.PORT || 8443 },
```

**Review by krysal — APPROVED** [review]
TIL this is possible. Nice! 💯

**Pull request merged**